matrix_sdk/src/async_client.rs

```diff
@@ -645,6 +646,25 @@ impl AsyncClient {
 
         let mut response = self.send(request).await?;
 
+        // when events change state updated signals to state store to update database
+        let mut updated = self.iter_joined_rooms(&mut response).await?;
+
+        if self.iter_invited_rooms(&response).await? {
+            updated = true;
+        }
+
+        if self.iter_left_rooms(&mut response).await? {
+            updated = true;
+        }
+
+        let mut client = self.base_client.write().await;
+        client.receive_sync_response(&mut response, updated).await?;
+
+        Ok(response)
+    }
+
+    #[inline]
```

Comment:
Why are we inlining this? Did you measure if it improves performance?

Reply:
No I'm not sure why I had that. Is it to soon to add some benchmarks maybe using [criterion-rs](https://github.com/bheisler/criterion.rs)?

Reply:
Yeah we probably shouldn't waste our time yet. Even then, avoiding a huge amount of events using filters is the best anyways.